### PR TITLE
User can create a recipe

### PR DESCRIPTION
**Pull request by elvitak**
### This PR contains:

- recipe creation form
- fixture files for successful recipe creation and with name missing
- visitAndAuthenticate cypress command
- add post call to api

PT story: https://www.pivotaltracker.com/story/show/181136163

@Mljungho @DefyCab @giacoletti

### CODE REVIEW

**Review by Mljungho — COMMENTED** [review]
Looks good, Don't really find anything to comment on. 

@giacoletti how about you?

File: cypress/integration/loggedInUserCanCreateARecipe.feature.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      }).as("create");
+      cy.visitAndAuthenticate();
+      cy.get("[data-cy=my-recipes]").click();
+      cy.get("[data-cy=create-recipe]").click();
+      cy.get("[data-cy=name-input]").type("Pancakes");
+      cy.get("[data-cy=instructions]").type("Mix them together. Bake");
+    });
+
+    it("is expected to create recipe", () => {
+      cy.get("[data-cy=submit-btn]").click();
```
Comment (giacoletti) [inline cypress/integration/loggedInUserCanCreateARecipe.feature.js:16]: we don't make user actions in a "it" block, this click action should be moved in the "before" block

File: cypress/integration/loggedInUserCanCreateARecipe.feature.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/* eslint-disable no-undef */
+describe("Recipe", () => {
+  before(() => {
+    cy.intercept("GET", "**api/recipes**", { body: { recipes: [] } });
```
Comment (giacoletti) [inline cypress/integration/loggedInUserCanCreateARecipe.feature.js]: You don't need the stars to make the intercept work. You may have something like this:
```
cy.intercept("GET", "/api/recipes", { body: { recipes: [] } });
```

File: cypress/integration/userCanLogin.feature.js [hunk 3]
```diff
@@ -1,9 +1,7 @@
 /* eslint-disable no-undef */
 describe("User can log in", () => {
   before(() => {
-    cy.intercept("GET", "/api/recipes", {
-      fixture: "recipesIndexResponse"
-    });
+    cy.intercept("GET", "*/api/recipes", { body: "" });
```
Comment (giacoletti) [inline cypress/integration/userCanLogin.feature.js]: This intercept doesn't work. You may write it this way:
```
cy.intercept("GET", "/api/recipes", { body: { recipes: [] } });
```

File: cypress/integration/loggedInUserCanCreateARecipe.feature.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  });
+  describe("can be created by logged in user", () => {
+    before(() => {
+      cy.intercept("POST", "**api/recipes**", {
```
Comment (giacoletti) [inline cypress/integration/loggedInUserCanCreateARecipe.feature.js]: The stars are not needed. You may intercept using this string "/api/recipes"

File: cypress/integration/loggedInUserCanCreateARecipe.feature.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    it("is expected to show a success message", () => {
+      cy.get("[data-cy=flash-message]").should(
+        "contain",
+        "Your recipe has been created"
+      );
+    });
+  });
+  describe("can't be created with empty name field", () => {
+    before(() => {
+      cy.intercept("POST", "**api/recipes**", {
```
Comment (giacoletti) [inline cypress/integration/loggedInUserCanCreateARecipe.feature.js]: Again, stars not needed. Intercept with "/api/recipes"

File: cypress/integration/loggedInUserCanCreateARecipe.feature.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        fixture: "createWithoutName.json"
+      });
+      cy.intercept("GET", "**api/recipes**", { body: { recipes: [] } });
```
Comment (giacoletti) [inline cypress/integration/loggedInUserCanCreateARecipe.feature.js]: Stars not needed

File: src/components/RecipeCreateForm.jsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        />
+      </div>
+      <Button variant="outlined" data-cy="submit-btn" onClick={createRecipe}>
+        Save
+      </Button>
+      {message ? (
+        <Alert data-cy="flash-message" severity="info">
+          {message}
+        </Alert>
+      ) : (
```
Comment (giacoletti) [inline src/components/RecipeCreateForm.jsx]: This "else" in the ternary operator is not needed. 
You may rewrite the conditional like:
```
{message && (
        <Alert data-cy="flash-message" severity="info">
          {message}
        </Alert>
   )}
```

File: src/modules/Recipes.js [hunk 8]
```diff
@@ -16,6 +16,19 @@ const Recipes = {
...
+  },
+  async create(recipe) {
+    try {
+      const response = await api.post("/recipes", {
+        recipe: {
+          name: recipe.name,
+          instructions: recipe.instructions
+        }
+      });
+      return response;
```
Comment (giacoletti) [inline src/modules/Recipes.js]: You want to return the "response.data" here. It's more manageable in the RecipeCreateForm.

File: src/components/RecipeCreateForm.jsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+import { TextField, Button, Alert } from "@mui/material";
+import Recipes from "../modules/Recipes";
+
+const RecipeCreateForm = () => {
+  const [recipe, setRecipe] = useState({});
+  const [message, setMessage] = useState();
+
+  const createRecipe = async () => {
+    const response = await Recipes.create(recipe);
+    setMessage(response.data.message);
```
Comment (giacoletti) [inline src/components/RecipeCreateForm.jsx]: When you return the "response.data" from the "Recipes.js", here you just need to set the message like:
```
setMessage(response.message)
```
This way you'll display either an error message or a successful message.

**Review by giacoletti — CHANGES_REQUESTED** [review]

File: cypress/integration/loggedInUserCanCreateARecipe.feature.js (hunk 1 above)
Comment (tochman) [reply cypress/integration/loggedInUserCanCreateARecipe.feature.js:16]: ...or at the very least, you need an assertion

**Pull request merged**